### PR TITLE
Group logs and show all task headers for plain output

### PR DESCRIPTION
**Pull request by eriwen**
### Context
We would like to group output for CI and piped output. See commit messages for how this is achieved.

FIXME: `ToolingApiIntegrationTest` is broken because a JVM is held open too long due the the executor service.

### Contributor Checklist
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass locally: `./gradlew quickCheck <impacted-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation including proper use of `@since` and `@Incubating` annotations for all public APIs

### CODE REVIEW

**Comment by w25r** [comment]
It doesn't appear that `ERROR` messages are grouped.  I get the following output:

```
--This is logged at ERROR

:first:logAtEveryLevel 
-----
-----
-----
Enumerating all error levels
Current log level is LIFECYCLE
isEnabled(DEBUG): false
isEnabled(INFO): false
isEnabled(LIFECYCLE): true
--This is logged at LIFECYCLE
isEnabled(WARN): true
--This is logged at WARN
isEnabled(QUIET): true
--This is logged at QUIET
isEnabled(ERROR): true
-----
-----
-----
```

It still groups properly when using rich console.

**Comment by w25r** [comment]
There seems to be a difference in behavior in the progress bar when running in rich console:
```
<========-----> 66% EXECUTING [7s]
> :second:logAtEveryLevel
> :second:logAtEveryLevel
```
This is different in both latest `release` and in the Daniel's long running output branch.  (Only one worker shows up.  For what it's worth the task logs some output, then sleeps for 10 seconds, and logs some more output.)

**Comment by w25r** [comment]
Task started and ended events not being grouped (#2044) is a bigger deal with this change.  Before, it was clear that output was not grouped.  Now when running in `--console=plain` it's difficult to determine if things are grouped incorrectly, or ungrouped.

File: subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/ProgressPlainLoggingIntegrationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.logging
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class ProgressPlainLoggingIntegrationTest extends AbstractIntegrationSpec {
```
Comment (big-guy) [inline subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/ProgressPlainLoggingIntegrationTest.groovy]: Do we also need to test interleaved tasks? like https://github.com/gradle/gradle/blob/656eec72102b23637285a23c966e08e015862e15/subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/BasicGroupedTaskLoggingFunctionalSpec.groovy#L119-L119

**Comment by bmuschko** [comment]
> We need to update the docs that Ben just added to call out that we do group tasks in parallel with `--console=plain`.

I already did that yesterday based on @eriwen's comment as he knew that it would change soon.

**Comment by big-guy** [comment]
@bmuschko good, I didn't check if that had already happened.

**Comment by bmuschko** [comment]
Closing as it is not going to be included in 4.0.

**Pull request closed without merging**